packages/helpers/helpers/merge-data-deep.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
```

Comment:
Can we use the existing and proven package [deepmerge](https://www.npmjs.com/package/deepmerge) for this instead of rolling our own helper? From experience, this can be a tricky beast to get right, so I'd rather use an existing package which is being downloaded and used a lot, rather than try to replicate and have to maintain this part of the logic ourselves.

In particular merging of arrays can be tricky, which this package handles very flexibly.